### PR TITLE
[expo cli] Added socket controls

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Add the ability to open the dev menu, reload an experience, and toggle element inspector, etc. from the CLI. This feature will only work after the new client is released with https://github.com/expo/expo/pull/12151 https://github.com/expo/expo/pull/12172 merged
- resolves https://github.com/expo/expo-cli/issues/160

# How

- Added Terminal UI commands (version gated to SDK 41):
  - `r` to reload any connected apps
  - `m` to open the dev menu in the client
  - `shift+m` to show a list of extra commands inspect elements, toggle perf monitor, reload, and open menu.
- Removed the ability to restart the bundler ('r', 'shift+r') from Terminal UI.
- "more tools" is disabled in "dev client" because there currently is no way to determine what commands are available on the native device, `reload`, and `devMenu` are baked into react-native core so those should continue to work.

# Test Plan

- iOS: Checkout https://github.com/expo/expo/pull/12151 and build client
- Android: Checkout https://github.com/expo/expo/pull/12172 and build client 
- Set project `sdkVersion` to `UNVERSIONED` (features should also be back dated later).
- Run `expo start`: 
  - press 'r' to reload apps
  - press 'm' to open menu
  - press 'm' again to close menu
  - press 'shift+m' to open more items
    - selecting 'Inspect Elements' should toggle element inspector.
    - selecting 'Performance Monitor' should toggle the perf monitor.
    - etc. with reload, and menu.
    - ctrl+c, or escape should cancel out of the select menu without throwing or ending the script process.
  - press 'm' with no devices connected, Metro should log a warning: `warn No apps connected. Sending "devMenu" to all React Native apps failed. Make sure your app is running in the simulator or on a phone connected via USB.`
- Changes should be able to run on physical devices as well (untested on iOS). 
- Run `expo start --dev-client`
  - `shift+m` should do nothing, the key binding shouldn't show up in the info log.

### CODE REVIEW

**Review by brentvatne — APPROVED** [review]
we should add something to indicate that these commands only work for ios and android

**Review by brentvatne — APPROVED** [review]

File: packages/expo-cli/src/commands/start/TerminalUI.ts [hunk 1]
```diff
@@ -317,7 +327,47 @@ export async function startAsync(projectRoot: string, options: StartOptions) {
...
+          Prompts.pauseInteractions();
+          try {
+            const value = await selectAsync({
+              // Options match: Chrome > View > Developer
+              message: `Expo Go tools ${chalk.dim`(native only)`}`,
+              choices: [
+                { title: 'Inspect elements', value: 'toggleElementInspector' },
+                { title: 'Performance monitor', value: 'togglePerformanceMonitor' },
+                { title: 'Developer menu', value: 'toggleDevMenu' },
+                { title: 'Reload native app', value: 'reload' },
```
Comment (EvanBacon) [inline packages/expo-cli/src/commands/start/TerminalUI.ts:361]:
```suggestion
                { title: 'Reload app', value: 'reload' },
```

File: packages/expo-cli/src/commands/start/TerminalUI.ts [hunk 2]
```diff
@@ -317,7 +331,47 @@ export async function startAsync(projectRoot: string, options: StartOptions) {
...
+          }
+
+          Prompts.pauseInteractions();
+          try {
+            const value = await selectAsync({
+              // Options match: Chrome > View > Developer
+              message: `Expo Go tools ${chalk.dim`(native only)`}`,
+              choices: [
+                { title: 'Inspect elements', value: 'toggleElementInspector' },
+                { title: 'Performance monitor', value: 'togglePerformanceMonitor' },
```
Comment (EvanBacon) [inline packages/expo-cli/src/commands/start/TerminalUI.ts:359]:
```suggestion
                { title: 'Toggle performance monitor', value: 'togglePerformanceMonitor' },
```

File: packages/expo-cli/src/commands/start/TerminalUI.ts [hunk 3]
```diff
@@ -317,7 +331,47 @@ export async function startAsync(projectRoot: string, options: StartOptions) {
...
+                { title: 'Developer menu', value: 'toggleDevMenu' },
```
Comment (EvanBacon) [inline packages/expo-cli/src/commands/start/TerminalUI.ts:360]:
```suggestion
                { title: 'Toggle developer menu', value: 'toggleDevMenu' },
```

**Pull request merged**